theano/gpuarray/opt.py

```diff
@@ -344,6 +380,9 @@ def apply(self, fgraph):
             for new_o, old_o in zip(outputs, node.outputs):
```

Comment:
To get the number of new nodes an optimization added, you must navigate the graph. The code: 

len(theano.gof.ops([mapping[i] for i in node.inputs], outputs)) 

should compute this.